discovery/kubernetes/endpointslice.go

```diff
@@ -338,6 +340,14 @@ func (e *EndpointSlice) buildEndpointSlice(eps endpointSliceAdaptor) *targetgrou
 			target[model.LabelName(endpointSliceAddressTargetNameLabel)] = lv(ep.targetRef().Name)
 		}
 
+		if ep.nodename() != nil {
```

Comment:
I think this is already added under another label if `attach_metadata.node: true` here https://github.com/prometheus/prometheus/blob/b938bbc11169dc224c103bec32a55daad0f3f34e/discovery/kubernetes/endpointslice.go#L347-L353

Reply:
Wouldn't it be confusing if we add it twice/ add it even if `attach_metadata.node: false`?

Reply:
`attach_metadata.node`: true implies that Prometheus watches the Node resource which requires additional permissions. The nice thing with endpointslices is that the node name is included in the endpointslices object.
Although if you need to filter targets on node labels/annotations, you'll still need to attach node metadata.

Reply:
BTW the "node name" labels have a `node_name` suffix while this PR uses `nodename`. I'll fix it to `node_name` to preserve consistency.